### ISSUE TITLE
Default initialization of modules

### ISSUE THREAD

**Issue by cgel** [torch/nn#449]
The default initialization of nn modules makes the gradients of very deep networks die out. And it makes the training slow or even impossible.
I've been looking in the documentation for information about initialization methods but i haven't found much.
How do you handle the initialization of your modules?
And where can i find more information?

**Comment by soumith** [comment]
You could look at this small script that implements different weight initialization methods: https://github.com/e-lab/torch-toolbox/blob/master/Weight-init/weight-init.lua

With Batch Normalization, even very deep networks seem to be stable and training fine, I'd suggesting adding those.
https://github.com/soumith/imagenet-multiGPU.torch/blob/master/models/alexnetowtbn_cudnn.lua

**Comment by cgel** [comment]
Thanks for that.
Do you know what method of initialization is used for constitutional and linear layers?

**Comment by kvitajakub** [comment]
From code it looks like caffe-xavier initialization.
https://github.com/torch/nn/blob/master/Linear.lua#L18
